specification/security.md

```diff
@@ -0,0 +1,241 @@
+          scopes:
+            board:read: Read the board
+            board:write: Write to the board
+```
+
+These can then be applied globally or for a given Operation, with a list of required scopes:
+
+```yaml
+openapi: 3.1.0
+info:
+  title: Tic Tac Toe
+  description: |
+    This API allows writing down marks on a Tic Tac Toe board
+    and requesting the state of the board or of individual squares.
+  version: 1.0.0
+security:
+  oauth2Profiles:
+    - board:read
+    - board:write
+paths:
+  /board:
+    get:
+      security:
+        oauth2Profiles: []
+```
+
+Note that if you want to segregate grant types - where, for example, Client Credentials is only supported for a specific Operation - you'll need to create a separate Security Scheme object that can be applied individually. This is also true if you want to differentiate the available scopes, for example:
+
+```yaml
+components:
+  securitySchemes:
+    app2AppOauth:
+      type: oauth2
+      flows:
+        clientCredentials:
+          tokenUrl: https://learn.openapis.org/oauth/2.0/token
+            # Only reading the board allow with delegated access
+            board:read: Read the board
+    user2AppOauth:
+        authorizationCode:
+          authorizationUrl: https://learn.openapis.org/oauth/2.0/auth
+          tokenUrl: https://learn.openapis.org/oauth/2.0/token
+          scopes:
+            # Reads and writes permitted via authorization code flow
+            board:read: Read the board
+            board:write: Write to the board
+```
+
+These then be declared as separate Security Requirements:
+
+```yaml
+info:
+  title: Tic Tac Toe
+  description: |
+    This API allows writing down marks on a Tic Tac Toe board
+    and requesting the state of the board or of individual squares.
+  version: 1.0.0
+
+paths:
+  /board:
+    get:
+      security:
+        app2AppOauth:
+        - board:read
+      ...
+  /board/{row}/{column}:
+    put:
+      security:
+        user2AppOauth:
+        - board:read
+        - board:write
+      ...
+```
+
+> Please refer to our [example OpenAPI document](../examples/tictactoe.yaml) for the complete example.
+
+## OpenID Connect
+
+The final Security Scheme type is OpenID Connect, which provides information for [OpenID Connect Discovery](https://openid.net/specs/openid-connect-discovery-1_0.html).
+
+OpenID Connect Core is obviously an OAuth 2.0 profile and is supported by some of properties the OAuth Flow Object. However, OpenID Connect is generally more complex than plain OAuth 2.0 and given OpenID Connect Discovery provides a machine-readable format at the discovery endpoint it makes sense to outsource this functionality entirely.
+
+Specifying OpenID Connect is therefore straightforward in that you provide the discovery endpoint in the property `openIdConnectUrl`:
+
+```yaml
+components:
+  securitySchemes:
+    openIdConnect:
+      type: openIdConnect
+      openIdConnectUrl: https://learn.openapis.org/.well-known/openid-profile
```

Comment:
should it be there https://learn.openapis.org/.well-known/openid-configuration 

rather like profile . 

following this 
https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationRequest

Reply:
Thanks @LasneF good shout 👍